great_expectations/profiler/domain_builder/simple_semantic_type_domain_builder.py

```diff
@@ -40,22 +38,19 @@ def _get_domains(
             ...
         ]
         """
-        # TODO: <Alex>ALEX -- How does this work?  What information is provided in **kwargs?  Can this be made explicit?  Where is "_type_filters" defined?</Alex>
         config: dict = kwargs
-        # TODO: AJB 20210416 If the type keyword for the DomainBuilder can contain multiple semantic types, should it be renamed types and take a list instead? Not that we can’t guess from what a user adds but something to make it clear that multiple semantic types can be used to construct a domain?
-        # TODO: <Alex>ALEX -- In general, to avoid confusion, we should avoid the use of "type" because it is a function in Python.</Alex>
-        type_filters: Union[str, Iterable, List[str]] = config.get("type_filters")
-        if type_filters is None:
-            # TODO: AJB 20210416 Add a test for the below comment - None = return all types
-            # None indicates no selection; all types should be returned
-            type_filters = self._type_filters
-        elif isinstance(type_filters, str):
-            type_filters = [self.SemanticDomainTypes[type_filters]]
-        elif isinstance(type_filters, Iterable):
-            type_filters = [self.SemanticDomainTypes[x] for x in type_filters]
+        semantic_types: Union[str, Iterable, List[str]] = config.get("semantic_types")
```

Comment:
Given that we name this variable semantic_types, does it make sense to have this variable take on a str value?

Reply:
@gilpasternak35 this came from the original prototype by James (and we use it in `DataConnector` too, perhaps in other places).  While I agree that it is not super clean, it is something we have been doing for simplicity for the users; so I solve this ambiguity by explicitly stating that its type can be a `Union` -- with that, I would feel it is acceptable.